app/server.R

```diff
@@ -699,59 +723,55 @@ server <- function(input, output, session) {
     ## Storing current target
     PREV_TARGET <<- input$targetVariable
 
-    ## do calling the api if we need as of data
-    ## otherwise we don't need to call it
+    ## Call the API only if we need as of data
     if (updateAsOf) {
       updateAsOfData(fetchDate = currentFetch)
     }
   })
 
-  observeEvent(input$scoreType,
-    {
-      if (input$targetVariable == "Deaths") {
-        df <- df %>% filter(signal == DEATH_FILTER)
-      } else if (input$targetVariable == "Cases") {
-        df <- df %>% filter(signal == CASE_FILTER)
-      } else {
-        df <- df %>% filter(signal == HOSPITALIZATIONS_FILTER)
-      }
-      # Only show forecasters that have data for the score chosen
-      updateForecasterChoices(session, df, input$forecasters, input$scoreType)
+  observeEvent(input$scoreType, {
+    if (input$targetVariable == "Deaths") {
+      df <- df %>% filter(signal == DEATH_FILTER)
+    } else if (input$targetVariable == "Cases") {
+      df <- df %>% filter(signal == CASE_FILTER)
+    } else {
+      df <- df %>% filter(signal == HOSPITALIZATIONS_FILTER)
+    }
+    # Only show forecasters that have data for the score chosen
+    updateForecasterChoices(session, df, input$forecasters, input$scoreType)
 
-      # If we are switching between coverage and other score types we need to
-      # update the as of data we have so it matches the correct locations shown
-      if (input$location == "US") {
-        updateAsOfData()
-      }
+    # If we are switching between coverage and other score types we need to
+    # update the as of data we have so it matches the correct locations shown
+    if (input$location == "US") {
+      updateAsOfData()
+    }
 
-      if (input$scoreType == "wis") {
-        showElement("wisExplanation")
-        hideElement("sharpnessExplanation")
-        hideElement("aeExplanation")
-        hideElement("coverageExplanation")
-      }
-      if (input$scoreType == "sharpness") {
-        showElement("sharpnessExplanation")
-        hideElement("wisExplanation")
-        hideElement("aeExplanation")
-        hideElement("coverageExplanation")
-      }
-      if (input$scoreType == "ae") {
-        hideElement("wisExplanation")
-        hideElement("sharpnessExplanation")
-        showElement("aeExplanation")
-        hideElement("coverageExplanation")
-      }
-      if (input$scoreType == "coverage") {
-        hideElement("wisExplanation")
-        hideElement("sharpnessExplanation")
-        hideElement("aeExplanation")
-        showElement("coverageExplanation")
-      }
-      USE_CURR_TRUTH <<- TRUE
-    },
-    ignoreInit = TRUE
```

Comment:
The `ignoreInit` flag keeps an `observe` from running when the dashboard is opened (since we have a bunch of inputs changing all at once, it can lead to lots of `observe`s being run unnecessarily). I re-enabled this one since the "about the scores" text wasn't rendering on load.